src/nvim/spell.c

```diff
@@ -2616,7 +2616,7 @@ static bool spell_mb_isword_class(int cl, win_T *wp)
   if (wp->w_s->b_cjk)
     // East Asian characters are not considered word characters.
     return cl == 2 || cl == 0x2800;
-  return cl >= 2 && cl != 0x2070 && cl != 0x2080;
+  return cl >= 2 && cl != 0x2070 && cl != 0x2080 && cl != 3;
```

Comment:
This is what the patch had but it can be simplified with ` cl > 3 && cl != 0x2070 && cl != 0x2080`